### ISSUE TITLE
Bug: Content pull fails using Sanity source when an ref array field is left empty

### ISSUE THREAD

**Issue by jasonwebb** [bluecadet/launchpad#58]
**Description:** On the Felician Sisters project I have some ref array fields that may be left blank. I found that when one of these fields is left blank, Launchpad fails to pull content for some reason.

**Expected behavior:** Running `npx launchpad content` should successfully pull down content from the Sanity source without errors.

**Actual behavior:** Running `npx launchpad content` produces the following console error:

> 2022-09-14-14:21:22 info: (content) Downloading 1 sources
> 2022-09-14-14:21:22 info: (content) Backing up 1 sources
> 2022-09-14-14:21:23 info: (content) Downloading source 1/1: content
> **2022-09-14-14:21:23 error: (content) Could not fetch page: Cannot read properties of undefined (reading '_type')**
> **2022-09-14-14:21:23 error: (content) Could not download all content: Cannot read properties of undefined (reading '_type')**
> 2022-09-14-14:21:23 info: (content) Restoring 1 sources
> 2022-09-14-14:21:23 info: (content) Restoring content from backup
> 2022-09-14-14:21:23 info: Launchpad exiting... :wave:
> 2022-09-14-14:21:23 info: Stopping apps...
> 2022-09-14-14:21:23 warn: All apps are already stopped
> 2022-09-14-14:21:23 info: ...launchpad shut down

**Steps to replicate:**

1. Create a field in your schema that accepts an array of refs, like so:

```
    {
      name: 'landing_ministries_stories',
      type: 'array',
      title: "Ministries Stories",
      group: 'landing',
      of: [
        {
          type: 'reference',
          to: [{type: 'story'}]
        }
      ]
    },
```
2. In Sanity Studio, leave this field empty.

3. Run `npx launchpad content`.

Note that if you go back to Sanity Studio and add one or more refs and retry `npx launchpad content`, it works as expected.

**How I worked around it:** 
I added a "dummy" ref with the title "EMPTY" for the field that needed to be empty. My code already silently skips docs that it can't find, so no additional work was necessary. But in other projects there might need to be additional code that conditionally skips doc lookups when EMPTY is found.

**Comment by benjaminbojko** [comment]
@pingevt do you have any time this week to look into this? Or @jasonwebb do you feel comfortable taking a look at the SanitySource class. Seems like it could be a simple fix.

**Comment by pingevt** [comment]
@benjaminbojko - Yup, I'm on it today.

**Comment by pingevt** [comment]
@jasonwebb - I can't reproduce this with just an empty field. Can you post your source config?

**Comment by jasonwebb** [comment]
See the first three fields under the Landing section in `province.json`:

[schemas.zip](https://github.com/bluecadet/launchpad/files/9575519/schemas.zip)

**Comment by pingevt** [comment]
sorry, @jasonwebb launchpad source config, pls?

**Comment by pingevt** [comment]
So... this is a combination of a few things, but the root is how we were looking for Sanity's structured data arrays. I applied the fix to the sanity updates branch.

If the value of the field is an array, it was checking if the `_type` of the first element was a "block". But, if the field is empty, there is no first element, hence the error. Also, it appeared we were applying all sanity text transforms regardless, which is not quite right. Updated that and added some warnings.

@jasonwebb 
I created a snapshot delivery. Can you try it out pls? `npm i @bluecadet/launchpad@0.0.0-felician-20220915142118`

@benjaminbojko - this is added to the #47 PR for updating Sanity Source.

**Comment by jasonwebb** [comment]
@pingevt switching to that package solved the issue! I confirmed by uninstall/reinstalling both versions of the package to replicate the issue, and it consistently went away when I used the above package.

**Comment by benjaminbojko** [comment]
Fixed in #47
